### ISSUE TITLE
Build failure with minimized build with pbes2 but without scrypt

### ISSUE THREAD

**Issue by vadz** [randombit/botan#1720]
I wrote a tiny utility creating a new RSA key and generating PKCS 10 certificate signing request for it, so I tried to configure Botan to contain only the functionality that I need using

```
./configure.py --minimized-build --enable-modules=x509,emsa_pkcs1,pbes2,rsa,sha2_32,system_rng
```

but this resulted in compilation errors due to `PasswordHashFamily` being undeclared when compiling `src/lib/pubkey/pbes2/pbes2.cpp`.

My understanding is that building should work for any combination of enabled modules (please ignore this issue if I'm wrong about it), so I think that either the code needs to be changed by adding more preprocessor checks or maybe `configure.py` needs to detect such incompatible modules combinations and e.g. add `scrypt` module automatically in this case (which is what I ended up doing).

This happens using a recent Git version (9fc0ad1e4ecc3725ee654ea51c9ee6bfae1537f8) under Windows with MSVC but I don't think this is OS/compiler-dependent.

**Comment by randombit** [comment]
> My understanding is that building should work for any combination of enabled modules 

Yes absolutely so, however sometimes we miss things, as here.

**Comment by randombit** [comment]
Fixed now in 759b45c, thank you for the report.

Though if you are generating+encrypting private keys it is worth it to use scrypt if you can.

**Comment by vadz** [comment]
Thanks for the fix! Yes, adding `scrypt` module was definitely the right thing to do and I did do it immediately (my full list of enabled modules needed to generate a new private key, save it after encrypting it with a passphrase and generate a CSR for it finally grew to `aes,cbc,emsa_pkcs1,pbes2,rsa,scrypt,sha2_32,system_rng,x509`), and I reported this small bug just in case anybody else runs into it.

Thanks again for ensuring this won't happen!
